src/Widgets/DataFormUploadWidget/DataFormUploadWidgetComponent.tsx

```diff
@@ -25,12 +25,37 @@ provideComponent(DataFormUploadWidget, ({ widget }) => {
   const [attachments, setAttachments] = useState<
     Array<{ file: File; key: string }>
   >([])
-  const [isTooLarge, setIsTooLarge] = useState(false)
+  const [rejectionErrors, setRejectionErrors] = useState<{
+    tooLarge: boolean
+    tooMany: boolean
+  }>({ tooLarge: false, tooMany: false })
 
-  const onDropAccepted = useCallback(() => setIsTooLarge(false), [])
-  const onDropRejected = useCallback(() => setIsTooLarge(true), [])
+  const onDropAccepted = useCallback(
+    () => setRejectionErrors({ tooLarge: false, tooMany: false }),
+    [],
+  )
+  const onDropRejected = useCallback((fileRejections: FileRejection[]) => {
+    const allErrors = fileRejections.flatMap((rejection) =>
+      rejection.errors.map((error) => error.code),
+    )
+    const unknownErrors = allErrors.filter(
+      (code) => !['file-too-large', 'too-many-files'].includes(code),
+    )
+
+    if (unknownErrors.length > 0) {
+      throw new Error(
+        `Unknown file rejection error codes: ${unknownErrors.join(', ')}`,
+      )
+    }
+
+    setRejectionErrors({
+      tooLarge: allErrors.includes('file-too-large'),
+      tooMany: allErrors.includes('too-many-files'),
+    })
+  }, [])
 
   const { getRootProps, getInputProps, inputRef, isDragActive } = useDropzone({
+    maxFiles: 10,
```

Comment:
* [ ] FYI In contrast to the input field limit this limit completely drops all selected files. To avoid frustration, it would be nice to accept as much input as possible (as does the input field) and only drop the files above the limit.